tests/integration/tier0/non-destructive/file-backup/test_file_backup.py

```diff
@@ -3,90 +3,151 @@
     """
-    os.environ["C2R_TESTS_MODIFIED_CONFIGS"] = "1"
+    backup_paths = {}
+
+    for pkg, config in PACKAGES.items():
+        file_name = config.rsplit("/", maxsplit=1)[-1]
+        # Install the packages if not installed already
+        if "is not installed" in shell(f"rpm -q {pkg}").output:
+            shell(f"yum install -y {pkg}")
+        # Create the config file, if not present already
+        if not os.path.exists(config):
+            shell(f"touch {config}")
+        # Backup the original file
+        shell(f"cp {config} {BACKUP_DIR}")
+        bkp_file_path = os.path.join(BACKUP_DIR, file_name)
+        # Remove the config to validate it won't get restored
+        # during the rollback
+        shell(f"rm -f {config}")
+
+        backup_paths[file_name] = [bkp_file_path, config]
 
     yield
 
-    del os.environ["C2R_TESTS_MODIFIED_CONFIGS"]
+    for file, paths in backup_paths.items():
+        bkp_file_path = paths[0]
+        default_config_path = paths[1]
+        # Verify the config did not get restored if not present
+        # prior the conversion
+        assert not os.path.exists(default_config_path)
+        # Restore the original file
+        assert shell(f"mv -f -v {bkp_file_path} {default_config_path}").returncode == 0
 
 
-@pytest.mark.parametrize("envar", [None, mod_config_envar])
+@pytest.mark.parametrize(
+    "file_action_fixture",
+    ["config_files_modified", "config_files_removed"],
+)
 @pytest.mark.test_file_backup
-def test_file_backup(convert2rhel, shell, envar, config_files):
+def test_file_backup(convert2rhel, shell, file_action_fixture, request):
     """
-    This test verifies correct handling backup and restore of config files.
-    Two configs (cloud-init, NetworkManager) are in scope of this test.
+    This test verifies correct handling of backup and restore of config files.
+    "/etc/cloud/cloud.cfg", "/etc/NetworkManager/NetworkManager.conf",
+    "/etc/logrotate.d/yum" and "/etc/logrotate.d/dnf" are in scope of this test.
     The following scenarios are verified:
     1/  The config files are modified with additional data
         The contents are compared pre- and post-conversion analysis task
         and should remain the same.
     2/  The config files are removed pre-conversion analysis task
         and should remain absent post-rollback.
+    Additionally, validates that a file clash does not happen during a backup.
+    There was a clash happening in the way convert2rhel backups files,
+    if there is a file to back up that has the same name as one directory
+    or another file already created, an error will be thrown to the user.
+    For that scenario we utilize "/etc/logrotate.d/yum"
+    and "/etc/logrotate.d/yum" files.
     """
+    # Create the backup directories
+    for directory in BACKUP_DIR, MODIFIED_FILES_DIR:
+        if not os.path.exists(directory):
+            shell(f"mkdir -v {directory}")
+    request.getfixturevalue(file_action_fixture)
```

Comment:
If it is possible move the dirs creation into the fixtures functions

Reply:
I guess I can move it to the fixtures, so they're standalone.